resources/views/checkout/steps/payment.blade.php

```diff
@@ -19,33 +19,33 @@ class="ml-auto w-8 h-8"
-        <div v-if="data" slot-scope="{ data }" class="mt-5">
-            <div v-for="agreement in data.checkoutAgreements">
-                <x-rapidez::slideover>
-                    <x-slot name="button">
-                        <a class="text-gray-700" href="#" v-on:click.prevent="toggle" v-if="agreement.mode == 'AUTO'">
-                            @{{ agreement.checkbox_text }}
-                        </a>
-                        <div v-else>
-                            <x-rapidez::checkbox
-                                name="agreement_ids[]"
-                                v-bind:value="agreement.agreement_id"
-                                v-model="checkout.agreement_ids"
-                                dusk="agreements"
-                                required
-                            >
-                                <a href="#" v-on:click.prevent="toggle">@{{ agreement.checkbox_text }}</a>
-                            </x-rapidez::checkbox>
-                        </div>
-                    </x-slot>
-
-                    <x-slot name="title">
-                        @{{ agreement.name }}
-                    </x-slot>
-
-                    <div v-if="agreement.is_html" v-html="agreement.content"></div>
-                    <div v-else v-text="agreement.content" class="whitespace-pre-line"></div>
+        <div slot-scope="{ data }" v-if="data" class="mt-5">
+            <template v-if="$root?.custom?.openAgreement">
+                <x-rapidez::slideover right id="checkoutAgreement">
+                    <x-slot:title>
+                        @{{ $root.custom.openAgreement.name }}
+                    </x-slot:title>
+                    <div v-if="$root.custom.openAgreement.is_html" v-html="$root.custom.openAgreement.content"></div>
+                    <div v-else v-text="$root.custom.openAgreement.content" class="whitespace-pre-line"></div>
                 </x-rapidez::slideover>
+            </template>
+            <div v-for="agreement in data.checkoutAgreements" :key="agreement.agreement_id">
+                <label class="text-gray-700 cursor-pointer underline hover:no-underline" for="checkoutAgreement" v-if="agreement.mode == 'AUTO'" v-on:click="window.Vue.set($root.custom, 'openAgreement', agreement)">
+                    @{{ agreement.checkbox_text }}
+                </label>
+                <div v-else>
+                    <x-rapidez::checkbox
+                        name="agreement_ids[]"
+                        v-bind:value="agreement.agreement_id"
+                        v-model="checkout.agreement_ids"
+                        dusk="agreements"
+                        required
+                    >
+                        <label class="text-gray-700 cursor-pointer underline hover:no-underline" for="checkoutAgreement" v-on:click="window.Vue.set($root.custom, 'openAgreement', agreement)">
```

Comment:
Vue was being a jerk about reactivity here, this was the only approach that worked.

Reply:
Do you know why reactivity wasn't functioning as expected?
Are you trying to set a string/null value to an object value without a deep?

I know vue is not always a fan of set via `window.app.custom`, even `$root` sometimes. But that is not necessary, it's just more verbose.

Reply:
I tried using:
- Regular assignment and Vue.set() on the `data` object we have, adding an extra custom key to use
- Same as above but also using `_renderProxy`
- Regular assignment and Vue.set() on a variable that I added in the slot-scope and then initialized via the classic `:set` hack
- Regular assignment and Vue.set() on `$root.custom`

Of all these, only the Vue.set() on $root.custom worked. Everything else didn't react properly and didn't have their values updated until you clicked a checkbox (which is very strange in and of itself). I tried to debug Vue itself with this but debugging Vue in-vivo is a pain and I didn't really get anywhere.

Reply:
I think i see your issue. this value is never defined on custom, so reactivity is never created for this nested value.

You should create it in https://github.com/rapidez/core/pull/464/files/0c8a29224561f26fc5cdfd2dbdd38284d1973b40#diff-251b99eac1abebedfe294a2334032b29b5f843d0b5d0834c32c055635f79c935R59 first